compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -484,3 +488,140 @@ private val PropertyDescriptor.declaresDefaultValue: Boolean
         is KtParameter -> declaration.defaultValue != null
         else -> false
     }
+
+fun <T : Annotation> KSAnnotated.getAnnotation(annotationKClass: KClass<T>): T? {
+    return this.annotations.firstOrNull {
```

Comment:
My concern here is on the [repeating annotations](https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html), in this case, you might need to think about whether returning the first instance of a repeating annotation is desired.

Meanwhile there is no such issue with `KSAnnotated.getAnnotationsByType(annotationKClass: KClass<T>)`

Reply:
@neetopia I was mirroring this API https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/AnnotatedElement.html#getAnnotation-java.lang.Class- for consistency

Reply:
Based on the [repeating annotations](https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html) documentation that you sent it also states "The behavior of the methods that return a single annotation, such as AnnotatedElement.getAnnotation(Class<T>), are unchanged in that they only return a single annotation if one annotation of the requested type is present. If more than one annotation of the requested type is present, you can obtain them by first getting their container annotation". So this should be satisfied by this behavior and consistent.

I was also keeping the KSAnnotated (KSP Kotlin) and AnnotatedElement (Java) as much in sync as possible with these methods for easy adoption and based on the Slack conversation that we had.

Reply:
I see, I didn't realize he was suggesting removal altogether, and not mirroring the Java API. That's fine I can understand that.